### PR TITLE
Use SHA256 to checksum Kubernetes binaries >= 1.17.0

### PR DESCRIPTION
**Pull request by tstromberg**
With the Kubernetes v1.18 release, SHA1 checksums are no longer provided.

### CODE REVIEW

**Comment by medyagh** [comment]
Needs linting.

**Comment by tstromberg** [comment]
PTAL. Lint issue was at master.

**Comment by tstromberg** [comment]
`TestStartStop/group/old-docker` appears to consistently fail. Fix incoming.

**Comment by afbjorklund** [comment]
@tstromberg : we do have a lint (govet) issue, about the fancy struct used in 9b1d2da4786f131bbbf038333ffcfd649409cde4:

`composite literal uses unkeyed fields`

File: pkg/minikube/machine/cache_binaries_test.go [hunk 1]
```diff
@@ -215,3 +218,57 @@ func TestCacheBinary(t *testing.T) {
...
+	var tc = []struct {
+		url     string
+		version string
+		want    download.FileOptions
+	}{
+		{
+			url:     "https://s/kubernetes-release/release/v1.16.0/bin/amd64/kubectl",
+			version: "v1.16.0",
+			want: download.FileOptions{
+				download.Options{
```
Comment (afbjorklund) [inline pkg/minikube/machine/cache_binaries_test.go]: Options:

File: pkg/minikube/machine/cache_binaries_test.go [hunk 2]
```diff
@@ -215,3 +218,57 @@ func TestCacheBinary(t *testing.T) {
...
+					Checksum:     "https://s/kubernetes-release/release/v1.16.0/bin/amd64/kubectl.sha1",
+					ChecksumHash: crypto.SHA1,
+				},
+				download.MkdirAll,
```
Comment (afbjorklund) [inline pkg/minikube/machine/cache_binaries_test.go]: Mkdirs:

**Comment by afbjorklund** [comment]
Would be nice to have support for k8s 1.18 in the new minikube release (1.8.0)

**Comment by afbjorklund** [comment]
It needs that lint fix first

**Pull request merged**